pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -30,33 +30,66 @@ import (
 
 // ClaimInfo holds information required
 // to prepare and unprepare a resource claim.
+// +k8s:deepcopy-gen=true
 type ClaimInfo struct {
-	sync.RWMutex
 	state.ClaimInfoState
 	// annotations is a mapping of container annotations per DRA plugin associated with
 	// a prepared resource
 	annotations map[string][]kubecontainer.Annotation
 	prepared    bool
 }
 
-func (info *ClaimInfo) addPodReference(podUID types.UID) {
-	info.Lock()
-	defer info.Unlock()
-
-	info.PodUIDs.Insert(string(podUID))
+// claimInfoCache is a cache of processed resource claims keyed by namespace + claim name.
```

Comment:
I thought this pair is uniq. Can you give some details?
I agree that using UID makes sense, just want to understand how the claim with the same name can be created in the same namespace.

Reply:
It's different incarnations under the same name. First "default/my-claim" gets created and prepared by kubelet. Then kubelet is shut down. A "sufficiently determined" user can now force-delete "default/my-claim", get it allocated again, and when kubelet comes back, "default/my-claim" is not the claim instance that was prepared before.

Reply:
This is not a simple change unfortunately. We don't always have the claimUID at the point at which we would like to do a lookup. I think we can get what we need, but it will likely take some restructuring of the cache itself. I'd rather do it as a follow up (or as a commit added on top once we have this core change reviewed).

Reply:
I'm fine with adding a TODO and a corresponding issue for tracking.

Reply:
Should `namespace + name` be changed to `namespace + "/" + name` already now? Who knows when we get around to using the UID instead.

Reply:
Yes. Let me do that.

Reply:
Updated along with reworking the commits a bit so they can be better reviewed 1-by-1